examples/CMakeLists.txt

```
@@ -20,6 +20,7 @@ add_subdirectory(cpp/custom_payload_pool)
 add_subdirectory(cpp/dds)
 add_subdirectory(cpp/delivery_mechanisms)
 add_subdirectory(cpp/discovery_server)
+add_subdirectory(cpp/security)
```

We should move it to the end of the file and change it to 
```suggestion
if(SECURITY)
    add_subdirectory(cpp/security)
endif()
```
In order to be built only if SECURITY is ON